core/module/registry.go

```diff
@@ -12,6 +13,7 @@ import (
 
 // Registry maintains a list of supported/enabled modules.
 type Registry struct {
+	sync.Mutex
```

Comment:
do not embed sync primitives. exposes the `Lock() / Unlock()` outside which if used externally can cause lot of tricky issues (including but not limited to deadlocks)

Reply:
Made the change, also using `sync.RWMutex` as suggested.